### ISSUE TITLE
Неверное склонение фамилии Кваша

### ISSUE THREAD

**Issue by tanraya** [petrovich/petrovich-rules#16]
Кваша Иван Иванович

```
род. Квашы Ивана Ивановича <--- WRONG
дат. Кваше Ивану Ивановичу
вин. Квашу Ивана Ивановича
тво. Квашой Иваном Ивановичем
пре. Кваше Иване Ивановиче
```

**Comment by mikhail-barg** [comment]
Сейчас родительный падеж склоняется правильно.
Зато есть неясность с творительным:

>Лемма Ожидалось Получилось 
>КВАША	КВАШЕЙ	КВАШОЙ	[:male, :instrumental]
>КВАША	КВАШЕЙ	КВАШОЙ	[:female, :instrumental]

Неясно, почему OpenCorpora предлагает вариант "Квашей", хотя на мой дилетантский взгляд ожидаемый результат как раз "КвашОй".

**Comment by ujifgc** [comment]
Та же песня с этими:

> АХТЫРЕЦ 	АХТЫРЦЕМ	АХТЫРЦОМ	[:male, :instrumental]
> ГРИШКОВЕЦ	ГРИШКОВЦЕМ	ГРИШКОВЦОМ	[:male, :instrumental]
> ГУБЕЦ	ГУБЦЕМ	ГУБЦОМ	[:male, :instrumental]
> ПРОКОПЕЦ	ПРОКОПЦЕМ	ПРОКОПЦОМ	[:male, :instrumental]
> СОСКОВЕЦ	СОСКОВЦЕМ	СОСКОВЦОМ	[:male, :instrumental]

Я думаю, здесь всё зависит от ударения: квАша — квАшей, квашА — квашОй. Как эту ситуацию разрулить кроме как через исключения — не знаю.

Насчёт их словаря — вчера писал на их почту про опечатки из https://github.com/petrovich/petrovich-eval/pull/2 — сегодня ответили и исправили оперативно.

**Comment by mikhail-barg** [comment]
Да, я тоже подумал про ударения, но в этом случае непонятно, как их разруливать в исключениях. Получается мы у себя можем зафиксировать только один вариант.

По опечаткам — спасибо, а то я уже себе внес в парсер их словаря принудительные замены по этим двум опечаткам.

**Comment by andrey2088** [comment]
Многие женские фамилии в родительном падеже склоняются неправильно.
Самый простой пример "Иванова".
Родительный падеж (кого? чего?) - "Иванову" ([пруф](https://www.analizfamilii.ru/Ivanova/skloneniye.html))
Но тут получаем "Ивановой", но это дательный падеж, а не родительный.
Вот тут в [коде](https://github.com/petrovich/petrovich-rules/blob/7672dff0a4a49402539ad60c0f115c96d384d058/rules.json#L339) должно быть "-у" вместо "-ой".
И так со многими женскими фамилиями.

**Comment by dustalov** [comment]
У базы правил нет координатора. Я буду рад принять PR, не ухудшающий качество склонения.

**Comment by xamgore** [comment]
@dustalov, а как у вас это качество считается? Есть какие-нибудь тесты?

**Comment by dustalov** [comment]
При каждом коммите и PR сюда запускаются тесты, в которых считается качество: https://travis-ci.org/github/petrovich/petrovich-rules.

**Comment by sszuev** [comment]
> 
> 
> Многие женские фамилии в родительном падеже склоняются неправильно.
> Самый простой пример "Иванова".
> Родительный падеж (кого? чего?) - "Иванову" ([пруф](https://www.analizfamilii.ru/Ivanova/skloneniye.html))
> Но тут получаем "Ивановой", но это дательный падеж, а не родительный.
> Вот тут в [коде](https://github.com/petrovich/petrovich-rules/blob/7672dff0a4a49402539ad60c0f115c96d384d058/rules.json#L339) должно быть "-у" вместо "-ой".
> И так со многими женскими фамилиями.

Здесь surnameonline.ru/surname.html `Иванова,Ивановой,Ивановой,Иванову,Ивановой,Ивановой`
думаю, все нормально тут.

**Comment by dustalov** [comment]
Пожалуйста, пришлите PR.

**Comment by nixprosoft** [comment]
Текущая версия Петровича склоняет правильно:
"Кто? Иванова Татьяна Михайловна"
"Нет кого? Ивановой Татьяны Михайловны"
"Кому? Ивановой Татьяне Михайловне"
"Вижу кого? Иванову Татьяну Михайловну"
"Кем? Ивановой Татьяной Михайловной"
"О ком? Ивановой Татьяне Михайловне"
Не вижу смысла в PR.

**Comment by nixprosoft** [comment]
Кваша сейчас вот так:

"Кто? Кваша Иван Иванович"
"Нет кого? Кваши Ивана Ивановича"
"Кому? Кваше Ивану Ивановичу"
"Вижу кого? Квашу Ивана Ивановича"
"Кем? Квашой Иваном Ивановичем"
"О ком? Кваше Иване Ивановиче"
